File: server/src/main/java/model/loot/LootFactory.java
package model.loot;

import model.item.IItem;
import model.item.Item;

import java.util.Random;

public class LootFactory implements ILootFactory {

    private static final Random rand = new Random();

    private IItem item;
    private int maxAmount;
    private double chance;

    public LootFactory(IItem item, int maxAmount, double chance) {
        this.item = item;
        this.maxAmount = maxAmount;
        this.chance = chance;
    }

    public IItem getItem() {
        return item;
    }

    public void setItem(IItem item) {
        this.item = item;
    }

    public int getMaxAmount() {
        return maxAmount;
    }

    public void setMaxAmount(int maxAmount) {
        this.maxAmount = maxAmount;
    }

    public double getChance() {
        return chance;
    }

    public void setChance(double chance) {
        this.chance = chance;
    }

    public Loot getLoot() {
        return getLoot(rand);
    }

    public Loot getLoot(Random random) {
        Loot loot = new Loot(item, 0);

        for (int i = 0; i < getMaxAmount(); i++) {
            if (random.nextDouble() < getChance()) {
                loot.addAmount(1);
            }
        }
        return loot;
    }
}

File: app/src/main/java/de/pnp/manager/app/model/LootFactory.kt
package de.pnp.manager.app.model

import de.pnp.manager.model.item.IItem
import de.pnp.manager.model.loot.ILoot
import de.pnp.manager.model.loot.ILootFactory
import java.util.*

class LootFactory : ILootFactory {
    override fun getItem(): IItem {
        TODO("Not yet implemented")
    }

    override fun setItem(item: IItem?) {
        TODO("Not yet implemented")
    }

    override fun getMaxAmount(): Int {
        TODO("Not yet implemented")
    }

    override fun setMaxAmount(maxAmount: Int) {
        TODO("Not yet implemented")
    }

    override fun getChance(): Double {
        TODO("Not yet implemented")
    }

    override fun setChance(chance: Double) {
        TODO("Not yet implemented")
    }

    override fun getLoot(): ILoot {
        TODO("Not yet implemented")
    }

    override fun getLoot(random: Random?): ILoot {
        TODO("Not yet implemented")
    }
}